From a29610f25da4ce4843be2df87fb644be87e6d35a Mon Sep 17 00:00:00 2001
From: Aleksander Morgado <aleksander@aleksander.es>
Date: Sun, 3 Sep 2017 13:01:42 +0200
Subject: [PATCH] plugin,port-probe: always cast SEND_DELAY value to guint64

    Program received signal SIGSEGV, Segmentation fault.
    strchr () at ../sysdeps/arm/armv6/strchr.S:28
    28    ../sysdeps/arm/armv6/strchr.S: No such file or directory.
    (gdb) bt
    #0  strchr () at ../sysdeps/arm/armv6/strchr.S:28
    #1  0x76b121c8 in g_param_spec_pool_lookup () from ~/buildroot/output/staging/lib/libgobject-2.0.so.0
    #2  0x76b0cf44 in g_object_new_valist () from ~/buildroot/output/staging/lib/libgobject-2.0.so.0
    #3  0x76b0d39c in g_object_new () from ~/buildroot/output/staging/lib/libgobject-2.0.so.0
    #4  0x75f75e40 in mm_plugin_create () at ublox/mm-plugin-ublox.c:99
    #5  0x00031550 in load_plugin (path=0xe9b68 "/usr/lib/ModemManager/libmm-plugin-ublox.so") at mm-plugin-manager.c:1521
    #6  load_plugins (error=0x7efffa68, self=0xdc4f0) at mm-plugin-manager.c:1574
    #7  initable_init (initable=<optimized out>, cancellable=<optimized out>, error=0x7efffa68) at mm-plugin-manager.c:1679
    #8  0x76b9b278 in g_initable_new_valist () from ~/buildroot/output/staging/lib/libgio-2.0.so.0
    #9  0x76b9b2e0 in g_initable_new () from ~/buildroot/output/staging/lib/libgio-2.0.so.0
    #10 0x000324d0 in mm_plugin_manager_new (plugin_dir=0xe9c40 "/usr/lib/ModemManager", error=error@entry=0x7efffa68) at mm-plugin-manager.c:1620
    #11 0x0002da08 in initable_init (initable=0xddb40, cancellable=<optimized out>, error=0x7efffa68) at mm-base-manager.c:1113
    #12 0x76b9b278 in g_initable_new_valist () from ~/buildroot/output/staging/lib/libgio-2.0.so.0
    #13 0x76b9b2e0 in g_initable_new () from ~/buildroot/output/staging/lib/libgio-2.0.so.0
    #14 0x0002e81c in mm_base_manager_new (connection=connection@entry=0xe1070, plugin_dir=plugin_dir@entry=0x9efb0 "/usr/lib/ModemManager",
        auto_scan=auto_scan@entry=1, initial_kernel_events=initial_kernel_events@entry=0x0, enable_test=0, error=0x7efffa68, error@entry=0x7efffa60)
        at mm-base-manager.c:957
    #15 0x0002b6ec in bus_acquired_cb (connection=0xe1070, name=<optimized out>, user_data=<optimized out>) at main.c:87
    #16 0x76c26864 in connection_get_cb () from ~/buildroot/output/staging/lib/libgio-2.0.so.0
    #17 0x76bc76cc in g_task_return_now () from ~/buildroot/output/staging/lib/libgio-2.0.so.0
    #18 0x76bc7d50 in g_task_return () from
    ~/buildroot/output/staging/lib/libgio-2.0.so.0

See similar bug in GStreamer:
https://bugzilla.gnome.org/show_bug.cgi?id=740191
---
 plugins/option/mm-plugin-hso.c      | 2 +-
 plugins/samsung/mm-plugin-samsung.c | 2 +-
 plugins/ublox/mm-plugin-ublox.c     | 2 +-
 src/mm-port-probe.c                 | 2 +-
 4 files changed, 4 insertions(+), 4 deletions(-)

diff --git a/plugins/option/mm-plugin-hso.c b/plugins/option/mm-plugin-hso.c
index de42c457..3a45ef38 100644
--- a/plugins/option/mm-plugin-hso.c
+++ b/plugins/option/mm-plugin-hso.c
@@ -186,7 +186,7 @@ mm_plugin_create (void)
                       MM_PLUGIN_ALLOWED_AT,         TRUE,
                       MM_PLUGIN_ALLOWED_QCDM,       TRUE,
                       MM_PLUGIN_CUSTOM_INIT,        &custom_init,
-                      MM_PLUGIN_SEND_DELAY,         0,
+                      MM_PLUGIN_SEND_DELAY,         (guint64) 0,
                       NULL));
 }
 
diff --git a/plugins/samsung/mm-plugin-samsung.c b/plugins/samsung/mm-plugin-samsung.c
index 9e6aaf94..e78f6895 100644
--- a/plugins/samsung/mm-plugin-samsung.c
+++ b/plugins/samsung/mm-plugin-samsung.c
@@ -65,7 +65,7 @@ mm_plugin_create (void)
                       MM_PLUGIN_ALLOWED_SUBSYSTEMS,  subsystems,
                       MM_PLUGIN_ALLOWED_PRODUCT_IDS, products,
                       MM_PLUGIN_ALLOWED_AT,          TRUE,
-                      MM_PLUGIN_SEND_DELAY,          0,
+                      MM_PLUGIN_SEND_DELAY,          (guint64) 0,
                       NULL));
 }
 
diff --git a/plugins/ublox/mm-plugin-ublox.c b/plugins/ublox/mm-plugin-ublox.c
index d4a29236..6ebd8711 100644
--- a/plugins/ublox/mm-plugin-ublox.c
+++ b/plugins/ublox/mm-plugin-ublox.c
@@ -75,7 +75,7 @@ mm_plugin_create (void)
                                     MM_PLUGIN_ALLOWED_VENDOR_STRINGS, vendor_strings,
                                     MM_PLUGIN_ALLOWED_AT,             TRUE,
                                     MM_PLUGIN_CUSTOM_AT_PROBE,        custom_at_probe,
-                                    MM_PLUGIN_SEND_DELAY,             0,
+                                    MM_PLUGIN_SEND_DELAY,             (guint64) 0,
                                     NULL));
 }
 
diff --git a/src/mm-port-probe.c b/src/mm-port-probe.c
index 640c319a..c6fa8ad0 100644
--- a/src/mm-port-probe.c
+++ b/src/mm-port-probe.c
@@ -1220,7 +1220,7 @@ serial_open_at (MMPortProbe *self)
 
         g_object_set (ctx->serial,
                       MM_PORT_SERIAL_SPEW_CONTROL,   TRUE,
-                      MM_PORT_SERIAL_SEND_DELAY,     (subsys == MM_PORT_SUBSYS_TTY ? ctx->at_send_delay : 0),
+                      MM_PORT_SERIAL_SEND_DELAY,     (guint64)(subsys == MM_PORT_SUBSYS_TTY ? ctx->at_send_delay : 0),
                       MM_PORT_SERIAL_AT_REMOVE_ECHO, ctx->at_remove_echo,
                       MM_PORT_SERIAL_AT_SEND_LF,     ctx->at_send_lf,
                       NULL);
-- 
2.13.1

